src/NuGet.Core/NuGet.Packaging/PackageFolderReader.cs

```diff
@@ -105,14 +105,19 @@ public override IEnumerable<string> GetFiles()
             var searchFolder = new DirectoryInfo(_root.FullName);
 
             foreach (var file in searchFolder.GetFiles("*", SearchOption.AllDirectories).
-                Where(p => !p.FullName.EndsWith(PackagingCoreConstants.NupkgExtension, StringComparison.OrdinalIgnoreCase)))
+                Where(p => !IsRootPackageFile(_root, p)))
```

Comment:
Like we discussed, it is probably better to avoid IsRootPackageFile and instead enumerate the root seperately from subfolders, so comparisons are not required